### ISSUE TITLE
XLA Optimizer Mixed Precision Support

### ISSUE THREAD

**Issue by aingo03304** [tensorflow/tensorflow#50981]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.5.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

I used autoclustering with optimization level 2.
```
export TF_DUMP_GRAPH_PREFIX="./xla_hlo"
export TF_XLA_FLAGS="--tf_xla_clustering_debug --tf_xla_auto_jit=2"
export XLA_FLAGS="--xla_dump_hlo_as_dot --xla_dump_to=./xla_hlo"
```

I found that when I have the `DynamicLossScale` in mixed-precision setting, the graph for loss scaling is appended to the original graph containing forward and backward propagations,  
and the remaining part of the optimizer is emitted as a separate graph.

So, my questions is

Why does the `DynamicScaleLoss` generate separated graphs in XLA?

**Will this change the current api? How?**
The API should not be changed because already full precision training is supported.

**Who will benefit with this feature?**
Anyone who wants to analyze the XLA HLO graph for their accelerator.

**Any Other info.**

**Comment by aingo03304** [comment]
I found that when I have `FixedLossScale` policy , the graph is not seperated. I think the problem is in `DynamicLossScale`.

**Comment by aingo03304** [comment]
https://github.com/tensorflow/tensorflow/issues/34047
Maybe related to this issue.

**Comment by aingo03304** [comment]
### Base code

```python
import os
import argparse
import tensorflow as tf
import tensorflow_datasets as tfds
import tensorflow.keras.mixed_precision.experimental as mixed_precision

model_name = [
    'ResNet50',
    'ResNet101',
    'ResNet152',
    'VGG16',
    'VGG19',
    'EfficientNetB0',
    'EfficientNetB1',
    'EfficientNetB2',
    'EfficientNetB3',
    'EfficientNetB4',
    'EfficientNetB5',
    'EfficientNetB6',
    'EfficientNetB7',
]

parser = argparse.ArgumentParser()
parser.add_argument('--learning-rate', type=float, default=0.1, help="LR for SGD")
parser.add_argument('--momentum', type=float, default=0.9, help="Momentum for SGD")
parser.add_argument('--batch-size', type=int, default=32, help="Batch size")
parser.add_argument('--epochs', type=int, default=10, help="# of epoch")
parser.add_argument('--model', type=str, default='ResNet50', help="model name")
parser.add_argument('--datadir', type=str, required=True, help="dataset dir")
args = parser.parse_args()

IMG_DIM = 224
policy = mixed_precision.Policy('mixed_float16')
mixed_precision.set_policy(policy)

print(f'Compute dtype: {policy.compute_dtype}')
print(f'Variable dtype: {policy.variable_dtype}')

def get_wrapped_model():
    input_shape = (IMG_DIM, IMG_DIM, 3)
    image_input = tf.keras.layers.Input(shape=input_shape)
    model_fn = getattr(tf.keras.applications, args.model)
    model = model_fn(
        include_top=True,
        weights=None,
        input_tensor=image_input,
        pooling='max',
        classes=1000)
    input_ = tf.keras.layers.Input(batch_shape=model.input_shape)
    output = model(input_)
    output = tf.keras.layers.Activation('linear', dtype='float32')(output)
    return tf.keras.Model(input_, output)

print("Loading datasets... This takes some time.")
train_dataset = tf.keras.preprocessing.image_dataset_from_directory(
    os.path.join(args.datadir, 'train'),
    labels="inferred",
    label_mode="int",
    class_names=None,
    color_mode="rgb",
    batch_size=args.batch_size,
    image_size=(IMG_DIM, IMG_DIM),
    shuffle=True,
    seed=42,
    validation_split=None,
    subset=None,
    interpolation="bilinear",
    follow_links=False,
)

model = get_wrapped_model()
model.compile(
    optimizer='sgd',
    loss='sparse_categorical_crossentropy',
    metrics=['accuracy'])

model.fit(train_dataset)
```

Regardless of experimental or stable, the mixed precision training has

**Comment by aingo03304** [comment]
The problem was when `DynamicScaleLoss` is enabled, few iterations spend some time to figure out if the gradient has underflowed value. After it converges, it emits weight update kernels properly.
